demo/index.js

```diff
@@ -1,77 +1,306 @@
+        if (haveDragFn && target.textContent.includes(`X`)) {
+            let img = new Image();
+            img.src = `dragimage.jpg`;
+            dataTransfer.setDragImage(img, img.width / 2, img.height / 2);
+        }
     }
-  }
 }
 
 function handleDragOver(evt) {
-  if (draggable) {
-    evt.preventDefault();
-    evt.dataTransfer.dropEffect = `move`;
-  }
+    if (draggableColumn) {
+        evt.preventDefault();
+        evt.dataTransfer.dropEffect = `move`;
+    }
 }
 
 function handleDragEnter({ target }) {
-  if (draggable) {
-    target.classList.add(`over`);
-  }
+    if (draggableColumn) {
+        target.classList.add(`over`);
+    }
 }
 
 function handleDragLeave({ target }) {
-  if (draggable) {
-    target.classList.remove(`over`);
-  }
+    if (draggableColumn) {
+        target.classList.remove(`over`);
+    }
 }
 
 function handleDragEnd() {
-  draggable = null;
-  cols.forEach((col) => col.classList.remove(`over`));
+    draggableColumn = null;
+    cols.forEach((col) => col.classList.remove(`over`));
 }
 
 function handleDrop(evt) {
-  if (draggable === null) return;
+    if (draggableColumn === null) return;
 
-  evt.stopPropagation();
-  evt.stopImmediatePropagation();
-  evt.preventDefault();
+    evt.stopPropagation();
+    evt.stopImmediatePropagation();
+    evt.preventDefault();
 
-  if (draggable !== this) {
-    swapDom(draggable, this);
-  }
+    if (draggableColumn !== this) {
+        swapDom(draggableColumn, this);
+    }
 }
 
-// https://stackoverflow.com/questions/9732624/how-to-swap-dom-child-nodes-in-javascript
-function swapDom(a, b) {
-  let aParent = a.parentNode;
-  let bParent = b.parentNode;
-  let aHolder = document.createElement(`div`);
-  let bHolder = document.createElement(`div`);
-  aParent.replaceChild(aHolder, a);
-  bParent.replaceChild(bHolder, b);
-  aParent.replaceChild(b, aHolder);
-  bParent.replaceChild(a, bHolder);
+function enableDragForTasks() {
+    /** @type {HTMLElement | null} */
+    let currentlyDraggedTask = null;
+
+    const tasks = document.querySelectorAll(`#tasks .task`);
+    for (let i = 0; i < tasks.length; i++) {
+        const task = tasks.item(i);
+        task.addEventListener("dragenter", () => {
+            if (currentlyDraggedTask === task) return;
+
+            task.classList.add("drag-over");
+        });
+        task.addEventListener("drop", (event) => {
+            event.preventDefault();
+            event.stopPropagation();
+            event.stopImmediatePropagation();
+
+            task.classList.remove("drag-over");
+
+            if (currentlyDraggedTask !== task) {
+                swapDom(currentlyDraggedTask, task);
+            }
+        });
+        task.addEventListener("dragleave", () => {
+            task.classList.remove("drag-over");
+        });
+        task.addEventListener("dragover", (event) => event.preventDefault());
+        task.addEventListener("dragstart", () => {
+            currentlyDraggedTask = task;
+        });
```

Comment:
You also don't need a block for single-line assignment:
```suggestion
        task.addEventListener("dragstart", () => (currentlyDraggedTask = task));
```